### ISSUE TITLE
Improve Segment Submission

### ISSUE THREAD

**Issue by SameenAhnaf** [libre-tube/LibreTube#4710]
### Describe your suggested feature

- [x] 1. Show buttons and menus in following points only if `Contribute to Sponsorblock` in settings is enabled. By default, it should be disabled.
- [ ] 2. Replace *Submit* menu icon with *Record* / *Stop* icon. While recording a segment, the icon should stay red and continue blinking.
- [ ] 3. Below comments, there should be a menu  `Segments` menu for editing and showing segments. Always arrange segments according to video player time.
- [ ] 4. On top right corner of unsubmitted segments, show `Cancel` and `OK` buttons. In case of submitted ones, show `Suggest Edits`
- [ ] 5. Show a row of buttons under each unsubmitted segment: *Preview*, *Inspect*, *Delete* and *Segment type* drop-down menu. Their functions should be similar to Sponsorblock browser extension. However, inspecting should auto-pause the video after finishing the segment.
- [x] 6. Show time in `1:32:24.005` format. As of now, place cursor on either start or end time to edit the segment by number pad.
- [ ] 7. Place *Edit* icons on the left and right of start and end time respectively. Allow scrolling up or down through hours, minutes and seconds similar to timer apps. However, the differences between consecutive milliseconds should be 10 milliseconds, not 1. 
- [ ] 8. Unless segments are submitted, show them  in white on seekbar. If no segment type is selected, always let them be auto-skipped for the contributor.
- [ ] 9. Show segment type with color above each submitted segmrnt and a row of buttons under each of them: *Like*, *Dislike*, *Lock* and *Skip action* drop-down menu. Locking segments should be only possible if a VIP Sponsorblock ID is entered in settings. Skip actions should include mute, manual skip, auto-skip and unskip. By default, skip action will be selected according to settings for each segment type. This drop-down is for custom skip options to any specific segment only for the contributor.
- [ ] 10. Allow marking mute segments. They could be directly included in settings and segment type drop-down menu like this:
(A) Sponsor (Mute)
(B) Unpaid/Self Promotion (Mute)
(C) Interaction Reminder (Mute)
These options should have both mute and skip options in settings drop-down menu.

- [ ] 11. Include a draft symbol for all unsubmitted and edited by the user segments.
- [ ] 12. Show manual skip toast 5 seconds earlier before the segment begins. Also, include `Pause video` button to the toast.
- [ ] 13. Under each segment, `Save this segment duration as draft` checkbox. This will let segments be pinned on the top of "Show segments" menu like this:
`+12:32.031 • Sponsor`
If any user is playing his video at 4:00.000 and clicks on this button, a segment starting from 4:00.000 and ending at 16.32.031 will be recorded with Sponsor type selected. This will remove the hassle of recording segments of same duration over and over again. If any user clicks on this option after finishing the video, mark the segment at the end.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Bnyro** [comment]
I agree with adding a preference to disable the SponsorBlock submit button, however I don't aim to work on anything else on that list as I don't use that feature.
PRs welcome.

**Comment by deonwaju** [comment]
Good day, I would like to contribute
